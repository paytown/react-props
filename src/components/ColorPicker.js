import React from 'react';

export default function ColorPicker() {
  const redHandler = () => {
    console.log('red');
  };
  const greenHandler = () => {
    console.log('green');
  };
  const blueHandler = () => {
    console.log('blue');
  };

  return (
    <section>
      <button onClick={redHandler}>Red</button>
      <button onClick={greenHandler}>Green</button>
      <button onClick={blueHandler}>Blue</button>
    </section>
  );
}
