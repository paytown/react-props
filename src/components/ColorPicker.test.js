import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('colorpicker', () => {
  it('display color picker', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('handles red button press', () => {
    console.log = jest.fn();
    const wrapper = shallow(<ColorPicker />);

    wrapper.find('button').at(0).simulate('click');

    expect(console.log).toHaveBeenCalledWith('red');
  });
});
