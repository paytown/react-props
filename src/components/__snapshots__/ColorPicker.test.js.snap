// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorpicker display color picker 1`] = `
<section>
  <button
    onClick={[Function]}
  >
    Red
  </button>
  <button
    onClick={[Function]}
  >
    Green
  </button>
  <button
    onClick={[Function]}
  >
    Blue
  </button>
</section>
`;
